Geography/Scottish Postcode Directory/3_Compare R and SPSS Outputs for SPD.R

```diff
@@ -4,89 +4,114 @@
 # Data release - Scotish Postcode Directory
 # Original Author - Calum Purdie
 # Original Date - 06/09/2018
-# Type - Check
+# Updated - 04/11/2019
+# Type - Comparison
 # Written/run on - R Studio Desktop 
 # Version - 3.5.1
 #
+# install.packages("magrittr")
 # install.packages("tidyr")
 # install.packages("dplyr")
 # install.packages("stringr")
 # install.packages("haven")
 # install.packages("sjlabelled")
+# install.packages("janitor")
+# install.packages("tidylog")
+# install.packages("glue")
 #
-# Description - This document is based on the SPSS syntax for comparing R and SPSS output for 
-#               Small Area Population Estimates found within GPD folders. It is designed to allow for the same data 
-#               checks to be made. Each section of the SPSS syntax is contained in this file within different
-#               subsections.
+# Description - Code for comparing R and SPSS files for 
+#               Scottish Postcode Directory
 #
 # Approximate run time - 5 minutes
 
 # Set working directory
-SPSS_filepath <- file.path("//Freddy", "DEPT", "PHIBCS", "PHI", "Referencing & Standards", "GPD", "1_Geography", "Scottish Postcode Directory", "Lookup Files")
+
+SPSS_filepath <- file.path("//Freddy", "DEPT", "PHIBCS", "PHI", 
+                           "Referencing & Standards", "GPD", "1_Geography", 
+                           "Scottish Postcode Directory", "Lookup Files")
 R_filepath <- file.path(SPSS_filepath, "R files")
 
 # Read in packages from library
+
+library(magrittr)
 library(tidyr)
 library(dplyr)
 library(stringr)
 library(haven)
 library(sjlabelled)
 library(janitor)
+library(tidylog)
+library(glue)
 
 ### 2 - Reading in files ----
 
 # Read in SPSS file and sort by pc7
 # Remove variable labels, formats and widths from SPSS
+# Haven reads in SPSS strings as factors
+# Convert all factors to characters in the SPSS file
 
-SPD_SPSS <- read_sav(file.path(SPSS_filepath, "Scottish_Postcode_Directory_2019_2.sav"), user_na=F) %>%
+SPD_SPSS <- read_sav(glue("{SPSS_filepath}/Scottish_Postcode_Directory_2019_2.sav"), 
+                     user_na=F) %>%
   arrange(pc7) %>%
   zap_formats() %>%
   zap_widths() %>%
-  remove_all_labels()
+  remove_all_labels() %>% 
+  mutate_if(is.factor, as.character)
```

Comment:
There is a way to automate it further. As the file name changes every 6 months, instead of manually changing it we can use the dir_ls function from fs package to auto detect the SPSS file (as there is only 1 SPSS file in the folder):

```
SPD_SPSS <- fs::dir_ls(glue("{SPSS_filepath}/"), regexp = ".sav$") %>%
  read_sav(user_na=F) %>% 
  arrange(pc7) %>%
  zap_formats() %>%
  zap_widths() %>%
  remove_all_labels() %>% 
  mutate_if(is.factor, as.character)
```